File: 1781-check-if-two-string-arrays-are-equivalent/check-if-two-string-arrays-are-equivalent.cpp
class Solution {
public:
    bool arrayStringsAreEqual(vector<string>& word1, vector<string>& word2) {
        int i = 0, j = 0;
        for(string &s: word1){
            for(char &c: s){
                if(i == word2.size() or word2[i][j] != c) return false;
                j++;
                if(j == word2[i].size()){
                    i++;
                    j = 0;
                }
            }
        }
        return i == word2.size();
    }
};

File: 1781-check-if-two-string-arrays-are-equivalent/check-if-two-string-arrays-are-equivalent.java
class Solution {
    public boolean arrayStringsAreEqual(String[] word1, String[] word2) {
        int i=0; 
        int j=0;

        for(String s:word1){
            for(char c:s.toCharArray()){
               if (i == word2.length || word2[i].charAt(j) != c) {
                   return false;
               }
               j++;
               if(j == word2[i].length()){
                   i++;
                   j=0;
               }
            }
        }
        return i == word2.length;
    }
}